### PR TITLE
Custom Welcome messages

### LINKED ISSUE #41: Custom Welcome Messages
Having a message that explains the server features and philosophy would be nice for first time visits. After that when you log in you would only see a short welcome back message

AC:
-players can choose exactly what their welcome screen message looks like by turning on/off available pieces of the message. EG: `!welcome autostore` makes it so i see the status of my autostore when i log in.
-store custom welcome settings in player database.
-when player changes message settings, they see a preview of their new welcome message.
-default is only the basic welcome message.

### PR DESCRIPTION
**Pull request by Gobsmack90**
here's what i came up with, if you think of any other things players may want to see upon opening the server let me know, otherwise its good to go as long as you approve.

closes #41

### CODE REVIEW

File: app/commands/welcome.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            { text: '!help welcome', color: 'green' },
+            ' for more info.\nThis includes a handful of hints aimed to help players discover new features.'
+        ];
+
+        this.helpShortDescription.welcome = [
+            'Customize the message that greets you when you login. ex: ',
+            { text: '!welcome cooldowns', color: 'green' },
+        ];
+
+        this.helpFullDescription.welcome = [
```
Comment (thorbaert) [inline app/commands/welcome.js:21]: This creates a dependency for every other file on this file. Not sure I like how this is setup

File: app/commands/welcome.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            'online': 'b',
+            'help': 'c',
+            'cooldowns': 'd',
+            'autostore': 'e',
+            'version': 'f',
+            'hints': 'z',
+            'reset': 'reset'
+        };
+
+        this.serverVersion = [
```
Comment (thorbaert) [inline app/commands/welcome.js]: For what purpose are we keeping a version and telling it to the user?
Versions are usually handled a lot different

File: app/data/index.js [hunk 3]
```diff
@@ -202,6 +202,45 @@ module.exports = Base => class extends Base {
...
+        //this applies to homes and shared homes
+        return new Promise((resolve) => {
+            Player.findOne({ name: playerName }, (err, player) => {
+                if (err) return reject(err);
+                resolve(player.welcome);            
+            });
+        });
+    }
+
+    updatePlayerWelcome(playerName, input) {
```
Comment (thorbaert) [inline app/data/index.js]: Too much logic in this. most all of this logic should live outside of this function.. Most anything in the data file should simply take what you've already worked out and add it

File: app/lib/hints.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
+module.exports = Base => class extends Base {
```
Comment (thorbaert) [inline app/lib/hints.js:1]: Challenge. 
Do you think you could figure out how to get this in the database?
Also when you pull it out of the database I believe you can query 1 random one.

**Review by thorbaert — CHANGES_REQUESTED** [review]

File: app/commands/welcome.js (hunk 2 above)
Comment (Gobsmack90) [reply app/commands/welcome.js]: removed the serverVersion stuff. I've replaced it with a xp store option. the fact i didn't originally include that was an oversight on my part.

File: app/commands/welcome.js (hunk 1 above)
Comment (Gobsmack90) [reply app/commands/welcome.js:21]: I've edited the description to make it much shorter. as for your worry about dependency i think that comes from a misunderstanding of what was communicated. the only things listed are options for the welcome message. functions separate from other files. you worried about adding future commands and how that would affect this, and the answer is that it wouldn't, unless we specifically decided we wanted a future welcome message option related to it.

File: app/lib/hints.js (hunk 4 above)
Comment (Gobsmack90) [reply app/lib/hints.js:1]: adding issue for this

**Review by thorbaert — APPROVED** [review]

**Pull request merged**